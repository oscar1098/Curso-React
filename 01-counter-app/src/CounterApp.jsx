import {useState} from 'react';
import PropTypes from "prop-types";


export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => {
        setCounter( counter + 1 )
    }

    const handleRest = () => {
        setCounter( counter-1 )
    }

    const handelRestart = () => {
        setCounter( value )
    }

    return ( 
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={  handleAdd }>
                +1
            </button>
            <button onClick={ handleRest } >-1</button>
            <button  aria-label='btn-reset' onClick={ handelRestart } >Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value : PropTypes.number.isRequired
}








