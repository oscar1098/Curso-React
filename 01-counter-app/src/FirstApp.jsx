import PropTypes from 'prop-types';

export const FirstApp = ({title, subTitle, name}) => {

  return (
    <>
    <div data-testid="test-title">{title}</div>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  );
}

FirstApp.propTypes = {
  name : PropTypes.string,
  subTitle : PropTypes.string,
  title : PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name : "Oscar Ardila",
  subTitle : "No hay subtitulo",
  // title : "No hay titulo",
}
