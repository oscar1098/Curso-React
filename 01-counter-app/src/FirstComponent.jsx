const newMessage = 'Oscar Ardila';

export const FirstComponent = () => {

    return (
        <>
            <h1> { newMessage } </h1>
            <h3>Subtitulo</h3>
        </>
    );
}

export const SecondComponent = () => {
    return (
        <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Autem sit asperiores nulla fugiat velit consequatur cupiditate, quibusdam possimus porro, facilis natus tenetur totam suscipit recusandae itaque similique aliquam animi repellendus?</p>

    )
}








