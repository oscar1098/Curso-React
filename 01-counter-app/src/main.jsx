import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstComponent, SecondComponent } from './FirstComponent';


ReactDOM.createRoot( document.getElementById('root') ).render(

    <React.StrictMode>
        <FirstComponent/>
        <SecondComponent/>
    </React.StrictMode>
);

