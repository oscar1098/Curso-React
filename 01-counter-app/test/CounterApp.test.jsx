import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Prueba en CounterApp', () => { 

    const valuei = 10;

    test('Debe hacer match con el snapchot', () => { 

        const { container } = render( <CounterApp value={valuei}/> );
        expect( container ).toMatchSnapshot()
     });

     test('Dede mostrar el valor inical de 100', () => { 

        render( <CounterApp value={100} /> );
        expect( screen.getByText(100)).toBeTruthy();

        // expect(screen.getByRole('heading',{level : 2}).innerHTML).toContain('100')
      });

      test('Debe incrementar con el boton +1', () => { 

        render( <CounterApp value={valuei}/> );
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('11') ).toBeTruthy();
       });
      
      test('Debe decrementar con el boton -1', () => { 

      render( <CounterApp value={valuei}/> );
      fireEvent.click( screen.getByText('-1') )
      expect( screen.getByText('9') ).toBeTruthy();
      
      });
      
      test('Debe reinicar con el boton reset', () => { 

      render( <CounterApp value={355}/> );
      fireEvent.click( screen.getByText('+1') )
      fireEvent.click( screen.getByText('+1') )
      fireEvent.click( screen.getByText('+1') )
      // fireEvent.click( screen.getByText('Reset') )
      fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))
      expect( screen.getByText( 355 ) ).toBeTruthy();


      });
 });