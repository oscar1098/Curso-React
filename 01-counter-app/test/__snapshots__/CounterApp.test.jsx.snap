// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en CounterApp Debe hacer match con el snapchot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
